store/store_test.go

```diff
@@ -81,9 +77,6 @@ func TestStore(t *testing.T) {
 
 	// check that the store can be successfully started after previous stop
 	// with all data being flushed.
-	store, err = NewStore[*headertest.DummyHeader](ds)
```

Comment:
Basically these 2 lines hid the stop-start bug from us. Here we were testing start of the new store based on `ds`. To fix that we just restart the existing store (and recreate `s.writesDn` in `.Start`, see `store.go`).